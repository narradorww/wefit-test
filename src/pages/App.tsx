import React from "react";
import List from "../components/List";
import Header from "../components/Header";
import EmptyCart from "../components/EmptyCart/index";
import Sucess from "../components/Sucess/index";

function App() {
  return (
    <div className="App">
      <Header />
      <List />
    </div>
  );
}

export default App;
