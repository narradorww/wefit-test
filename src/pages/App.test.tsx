import React from 'react';
import { render, screen } from '@testing-library/react';
import App from './App';

test('renders WeMovies on page', () => {
  render(<App />);
  const linkElement = screen.getByText(/WeMovies/i);
  expect(linkElement).toBeInTheDocument();
});

test('renders Meu Carrinho on page', () => {
  render(<App />);
  const linkElement = screen.getByText(/Meu Carrinho/i);
  expect(linkElement).toBeInTheDocument();
});

test('render a snapshot of the page', () => {
  const { container } = render(<App />);
  expect(container).toMatchSnapshot();
})

