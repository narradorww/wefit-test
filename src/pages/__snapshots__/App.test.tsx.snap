// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render a snapshot of the page 1`] = `
<div>
  <div
    class="App"
  >
    <div
      class="header"
    >
      <div
        class="logo"
      >
        <p>
          WeMovies
        </p>
      </div>
      <div
        class="cart"
      >
        <div
          class="wrapCart"
        >
          <div
            class="myCart"
          >
            <p>
              Meu Carrinho
            </p>
          </div>
          <div
            class="statusCart"
          >
            <p>
              0 itens
            </p>
          </div>
        </div>
        <div
          class="iconBasket"
        >
          <img
            alt="Icone de Cesta para Representar as Compras no Site"
            src="basket.svg"
          />
        </div>
      </div>
    </div>
    <section
      class="principal"
      data-testid="container"
    >
      <div
        class="card"
        data-testid="card"
      >
        <div
          class="image"
          data-testid="image"
        >
          <img
            alt="Capa do filme Viúva Negra"
            src="https://www.imagemhost.com.br/images/2022/07/10/viuva-negra.png"
          />
        </div>
        <div
          class="title"
          data-testid="title"
        >
          Viúva Negra
        </div>
        <div
          class="price"
          data-testid="price"
        >
          <p>
            R$ 
            9,99
          </p>
        </div>
        <div
          class="addCart"
          data-testid="button"
        >
          <img
            alt="Icone de um carrinho de mercado"
            src="cartplus.svg"
          />
          <p>
            0
          </p>
          <p>
            ADICIONAR AO CARRINHO
          </p>
        </div>
      </div>
    </section>
  </div>
</div>
`;
